src/pokemon.c

```diff
@@ -6977,3 +6978,40 @@ u32 CheckDynamicMoveType(struct Pokemon *mon, u32 move, u32 battler)
         return moveType;
     return gMovesInfo[move].type;
 }
+
+u32 GetRegionalForm(u32 species, u32 region)
+{
+    u32 formId = 0;
+    u32 firstFoundSpecies = 0;
+    const u16 *formTable = GetSpeciesFormTable(species);
+
+    if (formTable != NULL)
+    {
+        for (formId = 0; formTable[formId] != FORM_SPECIES_END; formId++)
+        {
+            if (firstFoundSpecies == 0)
+                firstFoundSpecies = formTable[formId];
+            
+            if ((gSpeciesInfo[formTable[formId]].isAlolanForm   && region == REGION_ALOLA)
+             || (gSpeciesInfo[formTable[formId]].isGalarianForm && region == REGION_GALAR)
+             || (gSpeciesInfo[formTable[formId]].isHisuianForm  && region == REGION_HISUI)
+             || (gSpeciesInfo[formTable[formId]].isPaldeanForm  && region == REGION_PALDEA))
```

Comment:
Since a near-identical check appears slightly below, is there any way these could be consolidated?

Reply:
Let me think of something...